tools/check.py

```diff
@@ -450,32 +521,49 @@ def validate(self):
 
 class IndexPageValidator(MarkdownValidator):
     """Validate the contents of the homepage (index.md)"""
-    HEADINGS = ['Topics',
+    EXPECTED_HEADINGS = ['Topics',
                 'Other Resources']
 
-    DOC_HEADERS = {'layout': vh.is_str,
-                   'title': vh.is_str}
-
-    CALLOUTS = {'prereq': ("Prerequisites", 1, 1)}
+    EXPECTED_YAML_HEADER = {'layout': vh.is_str,
+                            'title': vh.is_str}
 
-    def _partition_links(self):
-        """Check the text of every link in index.md"""
-        check_text = self.ast.find_external_links()
-        return check_text, []
+    EXPECTED_BOXES = {'prereq': ("Prerequisites", 1, 1)}
 
+    # TODO Improve the following function
     def _validate_intro_section(self):
-        """Validate the intro section
+        """Validate the intro section.
 
-        It must be a paragraph, followed by blockquoted list of prereqs"""
-        intro_block = self.ast.children[3]
-        intro_section = self.ast.is_paragraph(intro_block)
-        if not intro_section:
-            logging.error(
+        It must be a paragraph, followed by blockquoted with prereqs."""
+        if self.ast.body[0] and self.ast.body[0]['t'] != "Para":
```

Comment:
There are a lot of AST-specific operations being handled on the validator. In order to encourage separation of concerns, I'll do some refactoring to move this over to the helper class where possible.